### PR TITLE
Refactor builtin failsafe

### PR DESCRIPTION
**Pull request by nickanderson**
Refactor the builtin failsafe/bootstrap policy to make it more readable and
easier to maintain.

### CODE REVIEW

File: libpromises/failsafe.cf [hunk 1]
```diff
@@ -69,111 +92,172 @@ bundle agent failsafe_cfe_internal_update
...
+      # TODO: Decide if this class guard is appropriate. Should we
+      # guard checking of cf-execd process runnning to when inputs are
+      # repaired
+    !windows.inputdir_update_repaired::
 
-      "cf-execd" restart_class => "start_exec",
-      handle => "failsafe_cfe_internal_bootstrap_update_processes_start_cf_execd";
+      # We need to know when cf-execd is not running so that we can
+      # start it when necessary. Windows uses the service manager
+      # instead of keying on individual processes.
```
Comment (kacf) [inline libpromises/failsafe.cf]: FYI, systemd also does this, might be worth a note.

File: libpromises/failsafe.cf [hunk 2]
```diff
@@ -214,29 +308,39 @@ bundle agent failsafe_cfe_internal_update
        When updating masterfiles, wait (usually 5 minutes) for files to propagate to inputs on $(sys.policy_hub) before retrying."
       handle => "failsafe_cfe_internal_bootstrap_update_reports_did_not_get_policy";
 
-    server_started::
+    systemctl_restart_cfengine3_repaired::
+
+      "Restarted systemd unit cfengine3"
+        handle => "failsafe_cfe_intrnal_bootstrap_update_reports_systemd_unit_restarted";
+
+     systemctl_restart_cfengine3_error::
```
Comment (kacf) [inline libpromises/failsafe.cf]: Very minor: Indentation off by a space.

**Comment by kacf** [comment]
Good cleanup! Some issues, but all are minor. Just let me know what you plan to do.

File: libpromises/failsafe.cf (hunk 1 above)
Comment (nickanderson) [inline libpromises/failsafe.cf]: ack

File: libpromises/failsafe.cf (hunk 2 above)
Comment (nickanderson) [inline libpromises/failsafe.cf]: ack

**Comment by nickanderson** [comment]
@kacf thanks for the review, updated per your comments.

**Comment by kacf** [comment]
trigger build

**Comment by kacf** [comment]
Unless there is any serious issue fixed here, I think this should stay in master, it's quite risky for a patch release.

**Comment by kacf** [comment]
trigger build

**Comment by kacf** [comment]
trigger build

**Pull request merged**